### PR TITLE
Docs restructuring.

### PR DESCRIPTION
**Pull request by tomchristie**
Working through the docs refresh mentioned in https://github.com/encode/httpx/issues/947#issuecomment-1727389402.

This PR starts with some restructuring, without adding/removing/changing the content...

**Before**:

before

**After**:

after

### CODE REVIEW

**Review by karpetrosyan — APPROVED** [review]

**Comment by zanieb** [comment]
This is a great improvement!

File: docs/advanced/clients.md [hunk 1]
```diff
@@ -0,0 +1,326 @@
+# Client Instances
```
Comment (zanieb) [inline docs/advanced/clients.md:1]: Is this the only one with a title heading? Is this on purpose?

File: docs/advanced/clients.md [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+
+!!! hint
+    If you are coming from Requests, `httpx.Client()` is what you can use instead of `requests.Session()`.
+
+### Why use a Client?
```
Comment (zanieb) [inline docs/advanced/clients.md:6]: All of the other documents also use H2 headings instead of H3

**Comment by karpetrosyan** [comment]
I have found an issue when testing this locally.

Mkdocs says

```
WARNING -  Doc file 'index.md' contains a relative link 'advanced.md#calling-into-python-web-apps', but the target 'advanced.md' is not found among documentation files.
WARNING -  Doc file 'index.md' contains a relative link 'advanced.md', but the target is not found among documentation files.
WARNING -  Doc file 'async.md' contains a relative link './advanced.md#request-instances', but the target 'advanced.md' is not found among documentation files.
WARNING -  Doc file 'compatibility.md' contains a relative link 'advanced.md#mounting-transports', but the target 'advanced.md' is not found among documentation files.
WARNING -  Doc file 'compatibility.md' contains a relative link 'advanced.md#request-instances', but the target 'advanced.md' is not found among documentation files.
WARNING -  Doc file 'compatibility.md' contains a relative link 'advanced.md#client-instances', but the target 'advanced.md' is not found among documentation files.
WARNING -  Doc file 'compatibility.md' contains a relative link 'advanced.md#custom-transports', but the target 'advanced.md' is not found among documentation files.
WARNING -  Doc file 'environment_variables.md' contains a relative link 'advanced.md#http-proxying', but the target 'advanced.md' is not found among documentation files.
WARNING -  Doc file 'quickstart.md' contains a relative link 'advanced.md#fine-tuning-the-configuration', but the target 'advanced.md' is not found among documentation files.
WARNING -  Doc file 'third_party_packages.md' contains a relative link 'advanced.md#customizing-authentication', but the target 'advanced.md' is not found among documentation files.
WARNING -  Doc file 'third_party_packages.md' contains a relative link 'advanced.md#custom-transports', but the target 'advanced.md' is not found among documentation files.
WARNING -  Doc file 'troubleshooting.md' contains a relative link 'advanced.md#example', but the target 'advanced.md' is not found among documentation files.
WARNING -  Doc file 'troubleshooting.md' contains a relative link 'advanced.md#forward-vs-tunnel', but the target 'advanced.md' is not found among documentation files.
WARNING -  Doc file 'advanced/clients.md' contains a relative link 'quickstart.md', but the target 'advanced/quickstart.md' is not found among documentation files.
WARNING -  Doc file 'advanced/clients.md' contains a relative link 'http2.md', but the target 'advanced/http2.md' is not found among documentation files.
WARNING -  Doc file 'advanced/clients.md' contains a relative link 'quickstart.md', but the target 'advanced/quickstart.md' is not found among documentation files.
WARNING -  Doc file 'advanced/clients.md' contains a relative link 'api.md#client', but the target 'advanced/api.md' is not found among documentation files.
WARNING -  Doc file 'advanced/clients.md' contains a relative link 'api.md#request', but the target 'advanced/api.md' is not found among documentation files.
WARNING -  Doc file 'advanced/clients.md' contains a relative link 'img/tqdm-progress.gif', but the target 'advanced/img/tqdm-progress.gif' is not found among documentation files.
WARNING -  Doc file 'advanced/clients.md' contains a relative link 'quickstart.md#sending-multipart-file-uploads', but the target 'advanced/quickstart.md' is not found among documentation files.
WARNING -  Doc file 'advanced/proxies.md' contains a relative link 'troubleshooting.md#proxies', but the target 'advanced/troubleshooting.md' is not found among documentation files.
WARNING -  Doc file 'advanced/transports.md' contains a relative link 'environment_variables.md#http_proxy-https_proxy-all_proxy', but the target 'advanced/environment_variables.md' is not found among
           documentation files.
```

I will open a pull request that resolves this

**Comment by Kludex** [comment]
You can enforce warnings as errors on mkdocs - may be a good idea based on the above.

**Pull request merged**